addons/mail/static/src/core/common/message_model.js

```diff
@@ -78,7 +78,15 @@ export class Message extends Record {
     /** @type {boolean} */
     is_note;
     /** @type {boolean} */
-    is_transient;
+    is_transient = Record.attr(undefined, {
+        /** @this {import("models").Message} */
+        compute() {
+            if (this.is_transient !== undefined) {
+                return this.is_transient;
+            }
+            return !Number.isInteger(this.id);
+        },
+    });
```

Comment:
Do we really need this compute? Don't we just forget to set the transient flag when creating these specific messages?

Reply:
The `compute` has the advantage to work on all messages that are considered transient.
If we keep setting the flag by hand, we are chasing each occurrences in the code and this becomes fastidious to do.

This is not a getter because to follow stable policy, someone could have set the variable by hand, therefore I made it so that setting it by hand has precedence over the definition of a transient message based on non-integer id.

Reply:
I think I should add a 3rd concept:
- `isPending` is for temporary message being sent in channels
- `is_transient` is for receiving message transient notification to craft a transient message, with a specific look

Want I want here is actually tracking a non-genuine message so both `isPending` and `is_transient`.
IIRC livechat non persistent thread shows messages that are not flagged `isPending` nor `is_transient` but have float integers (because not a genuine thread and message until visitor posts). I was attempting to pick also these kinds of messages with the fix, but let me know if that's too much.

Reply:
The last solution is probably better for now. You can create a getter for the 3rd concept if it makes things easier. There are several places in the code where we check both together that can befinit from it.